### ISSUE TITLE
movementAnimations do not account for rotations on the vehicle

### ISSUE THREAD

**Issue by DrprofLuigi** [DonBruce64/MinecraftTransportSimulator#1230]
The `movementAnimations` on the Jerry can end up moving it up out of the angled mount on the Mule (part slot json below) because `movementAnimations` are not performed in relation to the rotation in the part slot.

```
		{
			"pos": [-0.5625, 0.125, 2.9375],
			"rot": [0, 0, 75],
			"types": ["interactable_jerrycan"],
			"defaultPart": "unuparts:unuparts_part_unu_barrel_jerrycan",
			"toneIndex": 1,
			"minValue": 0.0,
			"maxValue": 3.0
		}
```

This picture shows how it is supposed to sit (before I got fancy with `movementAnimations`)
![2022-02-13_18 01 55](https://user-images.githubusercontent.com/36775575/175852239-cfb4ce84-00e3-47ed-9fd4-180955f751e5.png)

This is how it sits with `movementAnimations`
![2022-06-26_20 02 19](https://user-images.githubusercontent.com/36775575/175852265-7594dec5-7983-4e08-829b-0d1734a16550.png)

**Comment by DonBruce64** [comment]
I'm a bit confused, what's going in the second picture?  I don't see a jerrycan in the first one, or are those boxes the jerrycans?

**Comment by DrprofLuigi** [comment]
They are mounted at an angle above the running board and under the body:

![image](https://user-images.githubusercontent.com/36775575/177694082-38420799-60bf-4b66-9317-c5cd27b65a7c.png)

**Comment by DonBruce64** [comment]
Fixed in V21.3.0.
